lib/mysql.js

```diff
@@ -72,31 +72,37 @@ function MySQL(settings) {
     s.connectionLimit = 10;
   }
 
-  var options = {
-    host: s.host || s.hostname || 'localhost',
-    port: s.port || 3306,
-    user: s.username || s.user,
-    password: s.password,
-    timezone: s.timezone,
-    socketPath: s.socketPath,
-    charset: s.collation.toUpperCase(), // Correct by docs despite seeming odd.
-    supportBigNumbers: s.supportBigNumbers,
-    connectionLimit: s.connectionLimit
-  };
-
-  // Don't configure the DB if the pool can be used for multiple DBs
-  if (!s.createDatabase) {
-    options.database = s.database;
-  }
-
-  // Take other options for mysql driver
-  // See https://github.com/strongloop/loopback-connector-mysql/issues/46
-  for (var p in s) {
-    if (p === 'database' && s.createDatabase) {
-      continue;
+  var options;
+  if (s.url) {
+    // use url to override other settings if url provided
+    options = s.url;
+  } else {
+    options = {
+      host: s.host || s.hostname || 'localhost',
+      port: s.port || 3306,
+      user: s.username || s.user,
+      password: s.password,
+      timezone: s.timezone,
+      socketPath: s.socketPath,
+      charset: s.collation.toUpperCase(), // Correct by docs despite seeming odd.
+      supportBigNumbers: s.supportBigNumbers,
+      connectionLimit: s.connectionLimit
+    };
+
+    // Don't configure the DB if the pool can be used for multiple DBs
+    if (!s.createDatabase) {
+      options.database = s.database;
```

Comment:
This snippet is inside `else` clause, not after, so it will not be executed if `options` is a string. 
I am aware of assigning var `options` a string value looks a little weird...I think change it to 
`mysql.createPool(url || options)`
would be more reasonable.

>>> For example, create a datasource with both `url` and the old settings, where the old settings are invalid, and then make sure that the connector connects to the database specified by `url`

As I explained in https://github.com/strongloop/loopback-connector-db2/pull/23#issuecomment-203496208
I verified this on my local for each connectors lol, then commit those PRs. I will think of a way to create the unit test for connection testing.